### PR TITLE
Adds CBMC proofs for session functions

### PR DESCRIPTION
**Pull request by tegansb**
*Description of changes:*
CBMC harnesses and Makefiles for aws_cryptosdk_session_set_commitment_policy, aws_cryptosdk_priv_algorithm_allowed_for_encrypt, aws_cryptosdk_priv_algorithm_allowed_for_decrypt. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: verification/cbmc/proofs/aws_cryptosdk_session_set_commitment_policy/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    * Called from aws_cryptosdk_priv_fail_session with the argument "new_state" hard-coded to ST_ERROR. 
+    * Only one case of the switch statement on "new_state" is therefore covered. 
+    * Additionally, the current state is not ST_CONFIG from a conditional check in aws_cryptosdk_session_set_commitment_policy. 
+
+Some functions are never reached:
+
+* `abort`
+
+    * abort is called from unexplored cases of aws_cryptosdk_priv_session_change_state.
```
Comment (feliperodri) [inline verification/cbmc/proofs/aws_cryptosdk_session_set_commitment_policy/README.md:18]:
```Markdown
# Memory safety proof for `aws_cryptosdk_session_set_commitment_policy`

This proof harness attains 40% code coverage.  The following comments explain
why the uncovered lines of code are unreachable code.

Some functions contain unreachable blocks of code:

* `aws_cryptosdk_priv_session_change_state`

    * Called from `aws_cryptosdk_priv_fail_session` with the argument `new_state` hard-coded to `ST_ERROR`. 
    * Only one case of the switch statement on `new_state` is therefore covered. 
    * Additionally, the current state is not `ST_CONFIG` from a conditional check in `aws_cryptosdk_session_set_commitment_policy`. 

Some functions are never reached:

* `abort()`

    * `abort()` is called from unexplored cases of `aws_cryptosdk_priv_session_change_state`.
```

**Review by feliperodri — CHANGES_REQUESTED** [review]

File: verification/cbmc/proofs/aws_cryptosdk_priv_algorithm_allowed_for_decrypt/cbmc-batch.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+build_memory: 32000
+cbmcflags: "--unwind;1;--flush;--object-bits;8;--external-sat-solver;kissat;--malloc-may-fail;--malloc-fail-null;--bounds-check;--conversion-check;--div-by-zero-check;--float-overflow-check;--nan-check;--pointer-check;--pointer-overflow-check;--pointer-primitive-check;--signed-overflow-check;--undefined-shift-check;--unsigned-overflow-check;--unwinding-assertions;--malloc-may-fail;--malloc-fail-null"
+coverage_memory: 32000
+expected: SUCCESSFUL
+goto: /Users/tegab/clean/aws-encryption-sdk-c/verification/cbmc/proofs/aws_cryptosdk_priv_algorithm_allowed_for_decrypt/gotos/aws_cryptosdk_priv_algorithm_allowed_for_decrypt_harness.c.goto
+jobos: ubuntu16
+property_memory: 32000
+report_memory: 32000
```
[the remark this answers is not in the document]
Comment (tegansb) [reply verification/cbmc/proofs/aws_cryptosdk_priv_algorithm_allowed_for_decrypt/cbmc-batch.yaml:8]: I see, I have replaced all yamls with the standard file!

File: verification/cbmc/proofs/aws_cryptosdk_priv_algorithm_allowed_for_encrypt/cbmc-batch.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+build_memory: 32000
+cbmcflags: "--unwind;1;--flush;--object-bits;8;--external-sat-solver;kissat;--malloc-may-fail;--malloc-fail-null;--bounds-check;--conversion-check;--div-by-zero-check;--float-overflow-check;--nan-check;--pointer-check;--pointer-overflow-check;--pointer-primitive-check;--signed-overflow-check;--undefined-shift-check;--unsigned-overflow-check;--unwinding-assertions;--malloc-may-fail;--malloc-fail-null"
+coverage_memory: 32000
+expected: SUCCESSFUL
+goto: /Users/tegab/clean/aws-encryption-sdk-c/verification/cbmc/proofs/aws_cryptosdk_priv_algorithm_allowed_for_encrypt/gotos/aws_cryptosdk_priv_algorithm_allowed_for_encrypt_harness.c.goto
+jobos: ubuntu16
+property_memory: 32000
+report_memory: 32000
```
[the remark this answers is not in the document]
Comment (tegansb) [reply verification/cbmc/proofs/aws_cryptosdk_priv_algorithm_allowed_for_encrypt/cbmc-batch.yaml:8]: Updated!

File: verification/cbmc/proofs/aws_cryptosdk_session_set_commitment_policy/cbmc-batch.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+build_memory: 32000
+cbmcflags: "--unwind;1;--flush;--object-bits;8;--external-sat-solver;kissat;--malloc-may-fail;--malloc-fail-null;--bounds-check;--conversion-check;--div-by-zero-check;--float-overflow-check;--nan-check;--pointer-check;--pointer-overflow-check;--pointer-primitive-check;--signed-overflow-check;--undefined-shift-check;--unsigned-overflow-check;--unwinding-assertions;--malloc-may-fail;--malloc-fail-null"
+coverage_memory: 32000
+expected: SUCCESSFUL
+goto: /Users/tegab/clean/aws-encryption-sdk-c/verification/cbmc/proofs/aws_cryptosdk_session_set_commitment_policy/gotos/aws_cryptosdk_session_set_commitment_policy_harness.c.goto
+jobos: ubuntu16
+property_memory: 32000
+report_memory: 32000
```
[the remark this answers is not in the document]
Comment (tegansb) [reply verification/cbmc/proofs/aws_cryptosdk_session_set_commitment_policy/cbmc-batch.yaml:8]: Updated!

File: verification/cbmc/proofs/aws_cryptosdk_session_set_commitment_policy/aws_cryptosdk_session_set_commitment_policy_harness.c [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+#include <aws/cryptosdk/session.h>
+#include <make_common_data_structures.h>
+
+void aws_cryptosdk_session_set_commitment_policy_harness() {
+    /* Nondet Input */
+    struct aws_cryptosdk_session *session = malloc(sizeof(*session));
+    enum aws_cryptosdk_commitment_policy policy;
+
+    /* Assumptions */
+    __CPROVER_assume(session != NULL);
```
[the remark this answers is not in the document]
Comment (tegansb) [reply verification/cbmc/proofs/aws_cryptosdk_session_set_commitment_policy/aws_cryptosdk_session_set_commitment_policy_harness.c:26]: We do now :) I forgot to push the commit adding session.c

File: verification/cbmc/proofs/aws_cryptosdk_session_set_commitment_policy/aws_cryptosdk_session_set_commitment_policy_harness.c [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+    /* Function under test */
+    int ret = aws_cryptosdk_session_set_commitment_policy(session, policy);
+    if (ret == AWS_OP_SUCCESS) {
```
[the remark this answers is not in the document]
Comment (tegansb) [reply verification/cbmc/proofs/aws_cryptosdk_session_set_commitment_policy/aws_cryptosdk_session_set_commitment_policy_harness.c:30]: Updated!

File: verification/cbmc/proofs/aws_cryptosdk_session_set_commitment_policy/README.md (hunk 1 above)
Comment (tegansb) [reply verification/cbmc/proofs/aws_cryptosdk_session_set_commitment_policy/README.md:18]: Updated!

**Review by alex-chew — APPROVED** [review]
LGTM

**Review by feliperodri — APPROVED** [review]

**Pull request merged**